flink-rpc/flink-rpc-akka-loader/src/main/java/org/apache/flink/runtime/rpc/akka/CleanupOnCloseRpcSystem.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rpc.akka;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.classloading.SubmoduleClassLoader;
+import org.apache.flink.runtime.rpc.AddressResolution;
+import org.apache.flink.runtime.rpc.RpcSystem;
+import org.apache.flink.util.Preconditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/** An {@link RpcSystem} wrapper that cleans up resources after the RPC system has been closed. */
+class CleanupOnCloseRpcSystem implements RpcSystem {
+    private static final Logger LOG = LoggerFactory.getLogger(CleanupOnCloseRpcSystem.class);
+
+    private final RpcSystem rpcSystem;
+    private final SubmoduleClassLoader pluginLoader;
+    private final Path tempFile;
+
+    public CleanupOnCloseRpcSystem(
+            RpcSystem rpcSystem, SubmoduleClassLoader pluginLoader, Path tempFile) {
+        this.rpcSystem = Preconditions.checkNotNull(rpcSystem);
+        this.pluginLoader = Preconditions.checkNotNull(pluginLoader);
+        this.tempFile = Preconditions.checkNotNull(tempFile);
+    }
+
+    @Override
+    public void close() {
+        rpcSystem.close();
+        try {
+            pluginLoader.close();
+        } catch (IOException e) {
+            LOG.warn("Could not close RpcSystem classloader.", e);
+        }
+        try {
+            Files.delete(tempFile);
+        } catch (IOException e) {
+            LOG.warn("Could not delete temporary rpc system file {}.", tempFile, e);
+        }
```

Comment:
I am wondering whether we shouldn't catch `Exception` in order to also catch `RuntimeException` during closing and continue with the clean up.